src/odemis/gui/comp/miccanvas.py

```diff
@@ -1165,6 +1166,110 @@ def update_drawing(self):
+    
+    def __init__(self, *args, **kwargs):
+        super(NavigableBarPlotCanvas, self).__init__(*args, **kwargs)
+        self.abilities |= {CAN_DRAG}
+        
+        # When resizing, margin to put around the current size
+        # TODO: Maybe make the margin related to the canvas size?
+        self.default_margin = 512
+        self.margins = (self.default_margin, self.default_margin)
+
+        # the position the view is asking to the next buffer recomputation
+        # in buffer-coordinates: = 1px at scale = 1
+        self._data_buffer = None
+        self.window_x = None
+        self.window_y = None
+        self.display_xrange = None
+        self.display_yrange = None
+        self.data_xrange = None
+        self.data_yrange = None
+
+    def set_1d_data(self, xs, ys, unit_x=None, unit_y=None, range_x=None, range_y=None):
+        if len(xs) != len(ys):
+            msg = "X and Y list are of unequal length. X: %s, Y: %s, Xs: %s..."
+            raise ValueError(msg % (len(xs), len(ys), str(xs)[:30]))
+        self._data_buffer = (numpy.array(xs), numpy.array(ys))
+        self.unit_x = unit_x
+        self.unit_y = unit_y
+        self.data_xrange = range_x
+        self.data_yrange = range_y
+        self.display_xrange = range_x
+        self.display_yrange = range_y
+        self.set_ranges(range_x, range_y)
+        
+    def fit_view_to_content(self, recenter=None):
+        # note - need to do this via an event to the viewport. Otherwise it doesn't
+        # set the axes properly
+        self.reset_ranges()
+
+    def reset_ranges(self):
+        if self._data_buffer is None:
+            return
+
+        self.set_ranges(self.data_xrange, self.data_yrange)
+
+    def set_ranges(self, x_range, y_range):
+
+        if self._data_buffer is None:
+            return
+
+        (xst , yst) = self._data_buffer
+
+        # Deal with x
+        lo, hi = x_range
+        # w = numpy.where((xst > lo) * (xst < hi))
+        lox = numpy.searchsorted(xst, lo, side="left")
+        hix = numpy.searchsorted(xst, hi, side="right")
+        xs = xst[lox:hix]
+        ys = yst[lox:hix]
+
+        # Add a few points onto the beginning and end of the array
+        # to prevent gaps in teh data from appearing
+        if lox > 0 and lox < len(xst):
+            if lox > 1:
+                xs = numpy.insert(xs, 0, lo)
+                ys = numpy.insert(ys, 0, yst[lox])
+            else:
+                xs = numpy.insert(xs, 0, lo)
+                ys = numpy.insert(ys, 0, yst[lox - 1])
+
+        if hix < (len(self._data_buffer) - 1) and hix >= 0:
+            xs = numpy.append(xs, hi)
+            ys = numpy.append(ys, yst[hix])
+
+        # Deal with Y
+        if y_range is not None:
+            lo, hi = y_range
+            ys = numpy.clip(ys, lo, hi)
+
+        self.display_xrange = x_range
+        self.display_yrange = y_range
+
+        # Resample data based on window size
+        if len(xs) > self.ClientSize.x * 10:
```

Comment:
If you're using the ClientSize, it means that probably this function (or a similar one) should called whenever the ClientSize changes. However, I don't see such code... so probably the display will go wrong on the window resize.